Sources/MapboxMaps/Style/Style+Localization.swift

```diff
@@ -0,0 +1,88 @@
+@_implementationOnly import MapboxCommon_Private
+
+/// Support for localization of labels
+extension Style {
+
+    /// This function creates an expression that will localize the `textField` property of a `SymbolLayer`
+    /// - Parameter locale: A `SupportedLanguage` based `Locale`
+    internal func localizeLabels(into locale: Locale) {
+
+        /// Get all symbol layers that are currently on the map
+        let symbolLayers = allLayerIdentifiers.filter { layer in
+            return layer.type == .symbol
+        }
+
+        /// Expression to be applied to the `SymbolLayer.textField`to localize the language
+        /// Sample Expression JSON: `["format",["coalesce",["get","name_en"],["get","name"]],{}]`
+        let replacement = "[\"get\",\"name_\(getLocaleValue(locale: locale))\"]"
+
+        let expressionCoalesce = try! NSRegularExpression(pattern: "\\[\"get\",\\s*\"(name_.{2,7})\"\\]",
+                                                          options: .caseInsensitive)
+        let expressionAbbr = try! NSRegularExpression(pattern: "\\[\"get\",\\s*\"abbr\"\\]",
+                                                      options: .caseInsensitive)
+
+        for layerInfo in symbolLayers {
+            do {
+                let tempLayer = try layer(withId: layerInfo.id) as SymbolLayer
+
+                if var stringExpression = String(data: try! JSONEncoder().encode(tempLayer.textField), encoding: .utf8),
+                   stringExpression != "null" {
+                    stringExpression.updateExpression(replacement: replacement, regex: expressionCoalesce)
+                    stringExpression.updateExpression(replacement: replacement, regex: expressionAbbr)
+
+                    // Turn the new json string back into an Expression
+                    let data = stringExpression.data(using: .utf8)
+                    let convertedExpression = try JSONDecoder().decode(Expression.self, from: data!)
+
+                    try updateLayer(withId: tempLayer.id) { (layer: inout SymbolLayer) throws in
+                        layer.textField = .expression(convertedExpression)
+                    }
+                }
+            } catch {
+                Log.warning(forMessage: "Error localizing textField for Symbol Layer with ID: \(layerInfo.id)", category: "Style")
+            }
```

Comment:
Considering that this block of code is going to be executed many times, we should think about the performance aspect of it..

Using `layer(withId:)` and `updateLayer(withId:)` would result in quite a few roundtrips to JSON and back. Instead could we drop down to the lower level api (i.e. `style.setStyleLayerProperty(..)`) ?